source/Engine.cpp

```diff
@@ -1070,6 +1073,9 @@ void Engine::Draw() const
 		string amount = to_string(it.second);
 		Point textPos = pos + textOff + Point(-font.Width(amount), 0.);
 		font.Draw(amount, textPos, isSelected ? selectedColor : unselectedColor);
+		double ammoIconClickCentreX = (ammoBox.Right() + ammoBox.Left()) / 2.;
+		double ammoIconClickCentreY = pos.Y() + ICON_SIZE / 2.;
+		ammoClickZones.emplace_back(Point(ammoIconClickCentreX, ammoIconClickCentreY), Point(80., 30.), it.first);
```

Comment:
Where/when does the ammoClickZones get cleared? By reading the code I get the impression that this vector is growing at each call to Draw().

Reply:
(Note; it is probably more efficient to calculate the clickzones when there is an actual click,since the ammoBox itself already acts as a "container clickzone".)

Reply:
I forgot to add a clear line. I'll see about calculating them on demand.

Reply:
Does ad532a2 resolve this?

Reply:
Yes, I believe it does, but what happens now is that the clickzones are generated and then immediately used for determining if the specific zone is clicked. This suggests that another optimization is possible that completely skips the zone generated (and just uses the coordinates).

Reply:
How would that work?
It currently creates a (temporary) rectangle with the coordinates and immediately checks if that contains the click point.
Seems like an easy way to do it. Maybe just doing what Rectangle::Contains() does with the coordinates directly here would probably be a bit more efficient, but this is only ever going to run when a person clicks inside the ammoBox.

Reply:
True, we will not need such improvement for performance. The reason to still optimize would be for readability.

What we know at this point is that the click was within the ammoBox, so I had expected that we only need to check the Y-coordinate (since the X-coordinates should already have been checked by the ammoBox).
But maybe both need to be checked due to the padding...

Reply:
There is definitely padding. Since we do have to check both coordinates, I don't see any issues with how it is now.

Reply:
@petervdmeer do you still think this needs changes? If so, what sort?

Reply:
Needing? No.

Could still be improved? Maybe, but let's keep that for a follow-up PR.